src/pages/index.js

```diff
@@ -21,80 +21,271 @@ import {
+
+const deleteImagePopup = new PopupWithConfirm(
+  deleteCardModalSelector,
+  handleCardDelete
 );
+deleteImagePopup.setEventListeners();
 
-cardListSection.renderItems();
+// function for delete card
 
-// //POPUP WITH FORM
+function handleCardDelete() {
+  deleteImagePopup.setLoading(true);
+  const id = deleteImagePopup.getId();
+  api
 
-const newCardPopup = new PopupWithForm(profileAddModal, submitCard);
-newCardPopup.setEventListeners();
+    .deleteCard(id)
+    .then((res) => {
+      console.log(res);
+      deleteImagePopup.remove();
+      deleteImagePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      deleteImagePopup.setLoading(false, "Yes");
+    });
+  deleteImagePopup.open(id);
+}
+
+//delete card validator instance
+
+const deleteCardFormValidator = new FormValidator(
+  settings,
+  deleteCardModalSelector
+);
+deleteCardFormValidator.enableValidation();
+
+//-----------------------------------------------------------------CARDS--------------------------------------------------------------------
 
 const cardOpenPopup = new PopupWithImage(cardOpenModal);
 cardOpenPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  profileEditModal,
-  handleEditFormSubmit
-);
-editProfilePopup.setEventListeners();
+function createCard(data) {
+  const newCard = new Card(
+    data,
+    userId,
+    "#card",
+    function handleCardClick(data) {
+      cardOpenPopup.open(data);
+    },
+
+    function handleDeleteClick() {
+      console.log(this);
+      deleteImagePopup.open();
+      deleteImagePopup.setId(data._id);
+      deleteImagePopup.setCard(newCard);
+    },
+
+    //card like count function
+
+    function handleCardLike(data) {
+      api
+        .changeLikeCardStatus(data._id, newCard.isLiked())
+        .then((res) => {
+          const likes = res.likes || [];
+          newCard.setLikes(likes);
+          newCard.toggleLikes();
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  );
+  return newCard.generateCard();
+}
+
+//-----------------------------------------------------ADD------------------------------------------------------------------------------------------------------------------------------
+
+//add form validator instance
 
-//PROFILE INFO
+const addCardFormValidator = new FormValidator(settings, profileAddModal);
+addCardFormValidator.enableValidation();
 
-const userInfo = new UserInfo(profileTitleSelector, profileDescriptionSelector);
+//calling new card popup
 
 profileAddButton.addEventListener("click", () => {
   addCardFormValidator.disableButton();
   newCardPopup.open();
 });
+//function add submit
+
+function submitCard({ title, url }) {
+  newCardPopup.setLoading(true);
+  api
+    .addCard(title, url)
+    .then((card) => {
+      const newCard = createCard(card);
+      cardListSection.prependItem(newCard);
+      newCardPopup.close();
+    })
+    .catch((err) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors